### ISSUE TITLE
Improve grunt task timing by requiring inside of tasks

### ISSUE THREAD

**Issue by zepumph** [phetsims/chipper#1007]
I want to experiement with the chipper and perennial gruntfiles by moving the require statement inside the task in which they are used. I think this will vastly improve the startup time for grunt, especially on windows. We could start by only doing this with modules only used by a single task (most). @samreid can you think of a reason why this is a bad idea? Here is an example of the type of commit I'm proposing. 

```diff
Index: js/grunt/Gruntfile.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/grunt/Gruntfile.js b/js/grunt/Gruntfile.js
--- a/js/grunt/Gruntfile.js	(revision 328eac7a4750b45d0e5a711a9c3408fe6f7c3b46)
+++ b/js/grunt/Gruntfile.js	(date 1612233999807)
@@ -8,9 +8,7 @@
 
 'use strict';
 
-const SimVersion = require( '../SimVersion' );
 const ChipperConstants = require( '../common/ChipperConstants' );
-const buildRunnable = require( './buildRunnable' );
 const buildStandalone = require( './buildStandalone' );
 const commitsSince = require( './commitsSince' );
 const generateA11yViewHTML = require( './generateA11yViewHTML' );
@@ -251,7 +249,7 @@
         for ( const brand of brands ) {
           grunt.log.writeln( `Building brand: ${brand}` );
 
-          await buildRunnable( repo, minifyOptions, instrument, allHTML, brand, localesOption, buildLocal );
+          await require( './buildRunnable' )( repo, minifyOptions, instrument, allHTML, brand, localesOption, buildLocal );
         }
       }
     } )
@@ -374,7 +372,7 @@
 
       // update README.md only for simulations
       if ( packageObject.phet.simulation && !packageObject.phet.readmeCreatedManually ) {
-        const simVersion = SimVersion.parse( packageObject.version );
+        const simVersion = require( '../SimVersion' ).parse( packageObject.version );
         generateREADME( repo, simVersion.isSimPublished );
       }
 

```

On average the chipper grunt file takes 2.5-3 seconds for me to load (before starting the actual task). When doing a task that forwards to perennial it adds between .5-1 second to then load the perennial file.

I found this by adding `process.exit()` at the beginning of a random grunt task, and then running something like `time grunt clean`.

I'm wondering though if there is something problematic about this that I'm not thinking through fully.

**Comment by samreid** [comment]
In some parts of perennial, there is a precedent where we dynamically require things, such as 

```js
  grunt.registerTask( 'pdom-comparison',
    'Compare two sim versions\' pdom',
    wrapTask( async () => {

      // Don't always require this, as we may have an older chipper checked out
      await require( './PDOMComparison' )( grunt.option( 'repo' ), grunt.option( 'sha' ) );
    } ) );
```

And a few places in the precommit hook.  

What if two tasks want to use the same require statement, should that be duplicated? Should we just find out which require statement is "the slow one" and either make it faster or make it the lazily `require`d one?  What if `grunt build` needs all or almost all of the require statements, and so cannot be very optimized?

I also think @jonathanolson would have some good recommendations in this area.

**Comment by samreid** [comment]
Another way to approach this problem: what grunt tasks are you running frequently that should be short but are taking too long because of all the require statements?  Maybe those should just be moved outside of grunt entirely.  Also, I thought I recalled another issue to get rid of grunt entirely but I can't find it at the moment.

**Comment by zepumph** [comment]
To me this feels like a numbers game. There are tens of grunt tasks, and no really giant ones (save build). My guess is that each require statement slows things down just a bit. That's a total guess though! Also perhaps requiring `grunt` is a large one, and this issue is a non-starter.

**Comment by jonathanolson** [comment]
> Also, I thought I recalled another issue to get rid of grunt entirely but I can't find it at the moment.

This was my first thought. If we're running node scripts, each one only requires what it needs, no? That's long-term, but we're not gaining much from grunt besides options-parsing and help-printing.

In the short-term... my immediate response is "if that saves time for a lot of tasks, that sounds worth it, but I'd like to understand the potential future drawbacks better"

**Comment by zepumph** [comment]
We do this now.
